languages/grammar/pt.json

```diff
@@ -0,0 +1,48 @@
+{
+    "meta": {
+        "regExpFlags": "gi"
+    },
+    "v5": {
+        "em": [
+            ["^ A(s?) ", " na$1 "],
+            ["^ O(s?) ", " no$1 "],
+            
+            ["^ (Marginal|Parte|Passagem|Ponte|Radial|Rede|Servidão) ", " na $1 "],
+            ["^ (Anel|Bosque|Boulevard|Cais|Canal|Caracol|Casal|Catete|Coronel|Duque|Fim|General|Impasse|Jardim|Juiz|Lote|Marechal|Mirante|Oriente|Parque|Professor|Setor|Tenente|Terminal|Túnel) ", " no $1 "],
+
+            ["^ ([^\\- ]+dade)(s?) ", " na$2 $1$2 "],
+            ["^ ([^\\- ]+z) ", " na $1 "],
+            ["^ ([^\\- ]+z)es ", " nas $1es "],
+            ["^ ([^\\- ]+[çz]ão) ", " na $1 "],
```

Comment:
I'm not sure for this case but also recommend `([^\\- ]+[çcz][ãa]o)`

Reply:
At a glance, the only false positive from loosening up this rule would be “[caos](https://en.m.wiktionary.org/wiki/caos)”, so I think we could do that.

Reply:
Updated expression doesn't match "caos" so as it looks for finishing whitespace after `o` :wink: